### ISSUE TITLE
When I try to add certain snippets of code which work well in regular markdown I get "undefined" errors

### ISSUE THREAD

**Issue by interglobalmedia** [rexxars/react-markdown#91]
The latest is specifically this snippet of code:
```
secondHand.style.mozTransform = `rotate(${secondsDegrees}deg)`;
secondHand.style.webkitTransform = `rotate(${secondsDegrees}deg)`;
secondHand.style.transform = `rotate(${secondsDegrees}deg)`;
```

I am being told in console that secondsDegrees is not defined:
```
Uncaught ReferenceError: secondsDegrees is not defined
```

This is supposed to be a snippet of code in markdown, so it should not matter whether or not something has been "defined".  I have been using Chrome, and I also use Atom and eslint. This kind of thing has happened several times already. How can I avoid it? Or can I? Thanks.

**Comment by interglobalmedia** [comment]
I have been testing this further, and it seems that react-markdown does NOT recognize some if not all es6 syntax. That is why I had the issue with this snippet. When I tried to fix it, eslint said I had to make all these other changes (especially to back ticks) that were ok before. This is a HUGE problem for those of us that use es6 and beyond. FYI. Thanks.

**Comment by rexxars** [comment]
This isn't a problem with react-markdown, but rather with how your code defines the markdown.

Let's image you want the following markdown to be rendered: 

    ## Template literals
    
    ```js
    const color = "blue"
    const str = `The car is ${color}`
    console.log(str)
    ```
    
    Variable interpolation is **neat**, right?

If you put this into a Javascript file like this:

    const markdown = `
    ## Template literals
    
    ```js
    const color = "blue"
    const str = `The car is ${color}`
    console.log(str)
    ```
    
    Variable interpolation is **neat**, right?`
    
It will give an error like the one you are encountering:
```
ReferenceError: markdown is not defined
```

This is because the backticks in this markdown will be interpreted as closing/starting ES6 template literals.

So in order to make this work, you need to escape the backticks by prefixing them with a `\` character. Having done that, you will also have to escape variable interpolation in the same way, escaping the `$` character. Altogether:

    const markdown = `
    ## Template literals

    \`\`\`js
    const color = "blue"
    const str = \`The car is \${color}\`
    console.log(str)
    \`\`\`

    Variable interpolation is **neat**, right?`

Note that this is only necessary because of the way the markdown is loaded/defined. If you were to define the same markdown with plain old ES5 string semantics, it would work just fine:

    const markdown = [
      '## Template literals',
      '',
      '```js',
      'const color = "blue"',
      'const str = `The car is ${color}`',
      'console.log(str)',
      '```',
      '',
      'Variable interpolation is **neat**, right?`',
    ].join('\n')

    console.log(markdown)

Or if you were fetching the markdown from a remote source, it would also just work:

```js
fetch('some-markdown-file.md')
  .then(res => res.text())
  .then(markdown => console.log(ReactDOM.renderToString(<Markdown source={markdown} />)))
```

The main takeaway from this is that the error you are seeing is a result of the input in and of itself and is not really related to react-markdown. If you take the same code that defines the markdown source and remove any reference to react-markdown, it'll give you the same error.

Hope this was helpful :)

**Comment by interglobalmedia** [comment]
Thanks @rexxars ! Yes, I found all that out the hard way. And honestly, just as you say, all that escaping just is not necessary. The only thing is that I began using react-markdown because I wanted to use it the re-haul of my portfolio site which I am building with React. I have a lot of articulation of my code/projects and code development related articles there as well, and I wanted to stay true to the code I was using in the projects as I had done in my previous portfolio site. I guess I will just have to add a disclaimer wherever necessary. I ended up using the Babel transpiler/repl on their site to transpile my es6 to es5 and then it worked fine. Thanks for all your help! Otherwise, am really loving this plugin thus far. It WILL save me much time in the future when I really have it down. But it's worth learning well for that reason. Thanks!
